metaflow/task.py

```diff
@@ -463,6 +471,37 @@ def run_step(
             # 3. initialize foreach state
             self._init_foreach(step_name, join_type, inputs, split_index)
 
+            # Add foreach stack to metadata of the task
+
+            foreach_stack = (
```

Comment:
I've been thinking about this and trying to find a way to do this without actually storing a potentially large (and worse, unbounded) string in the metadata. I think we can probably do the following. In the metadata, we can store the split index instead. The `_foreach_stack` is actually available above as well as the `_foreach_values` so we would then be able, in the UI, to reconstruct the string that is being constructed below. This would be a tad be more complicated in the UI (but not that much and it would have the added benefit of being able to use the cache so fewer requests (ie: for each foreach nest, we would only need to load 2 artifacts to have all the paths for all the tasks in the foreach).

Thoughts?

Reply:
Chatted during meeting, This could be achieved but may be overkill. We also want people to be able to more easily access the foreach path, so putting them in metadata also makes sense.

I added a limit to how much data we store in metadata. If we encounter the arbitrary large string problem we can revisit this in the future.

Reply:
Yep -- and now that w have a limit of characters this should be fine.